In [14]:
import pandas as pd

Q1 - Create a data frame with details of 10 students and columns as Roll Number, Name, Gender, Marks1, Marks2, Marks3. 

In [15]:
import numpy as np
data = {'Name':['Josh','Raj','Manoj','Tony','John','Toff','Iowa','Polly','Vardhan','Natasha'],'Roll No':range(1,11),'Gender':np.random.randint(2, size=10),'Marks1':np.random.randint(0,101,dtype=int,size=10),'Marks2':np.random.randint(0,101,dtype=int,size=10),'Marks3':np.random.randint(0,101,dtype=int,size=10)}
df = pd.DataFrame(data)
df

Name  Roll No  Gender  Marks1  Marks2  Marks3
0     Josh        1       0      73      38      25
1      Raj        2       0      74      90      92
2    Manoj        3       1      50      48      77
3     Tony        4       0       1      14      24
4     John        5       1      79      95      50
5     Toff        6       1      61      47      49
6     Iowa        7       1      82      65      18
7    Polly        8       1      80      89      40
8  Vardhan        9       1      55      12      74
9  Natasha       10       1       7      14      60

Q2 - Create a new column with total marks 

In [16]:
df['Total_Marks'] = df.Marks1+df.Marks2+df.Marks2
df

Name  Roll No  Gender  Marks1  Marks2  Marks3  Total_Marks
0     Josh        1       0      73      38      25          149
1      Raj        2       0      74      90      92          254
2    Manoj        3       1      50      48      77          146
3     Tony        4       0       1      14      24           29
4     John        5       1      79      95      50          269
5     Toff        6       1      61      47      49          155
6     Iowa        7       1      82      65      18          212
7    Polly        8       1      80      89      40          258
8  Vardhan        9       1      55      12      74           79
9  Natasha       10       1       7      14      60           35

Q3- Find the lowest marks in Marks1 

In [17]:
df.loc[df.Marks1==df.Marks1.min()]#Tony got lowest marks in Marks1=1

Name  Roll No  Gender  Marks1  Marks2  Marks3  Total_Marks
3  Tony        4       0       1      14      24           29

Q4- Find the Highest marks in Marks2 

In [18]:
df.loc[df.Marks2==df.Marks2.max()]#John got highest in Marks2 = 95

Name  Roll No  Gender  Marks1  Marks2  Marks3  Total_Marks
4  John        5       1      79      95      50          269

Q5-Find the average marks in Marks3 

In [19]:
df.Marks3.mean()#Average marks in  Marks3 given the markas of 10 students

50.9

Q6-Find student name with highest average 

In [20]:
df['Average_Marks'] = df['Total_Marks']/3
df.loc[df.Average_Marks==df.Average_Marks.max()]

Name  Roll No  Gender  Marks1  Marks2  Marks3  Total_Marks  Average_Marks
4  John        5       1      79      95      50          269      89.666667

John has the best average among the 10 students = 89.666667

Q7 - Find how many students failed in Marks2 (<40) 

In [21]:
df.loc[df.Marks2<40]

Name  Roll No  Gender  Marks1  Marks2  Marks3  Total_Marks  \
0     Josh        1       0      73      38      25          149   
3     Tony        4       0       1      14      24           29   
8  Vardhan        9       1      55      12      74           79   
9  Natasha       10       1       7      14      60           35   

   Average_Marks  
0      49.666667  
3       9.666667  
8      26.333333  
9      11.666667

4 students failed Marks2

COMPLETE